x-pack/platform/plugins/shared/onechat/public/application/components/tools/esql/edit_esql_tool.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { ToolFormProvider } from '../../../context/tool_form_provider';
+import { useEditTool } from '../../../hooks/tools/use_edit_tools';
+import { EsqlTool } from './esql_tool';
+import { OnechatEsqlToolFormMode } from './form/esql_tool_form';
+
+export const EditEsqlTool: React.FC = () => {
+  const { toolId } = useParams<{ toolId: string }>();
+
+  const {
+    tool: editingTool,
+    isSubmitting,
+    isLoading,
+    editTool,
+  } = useEditTool({
+    toolId,
+  });
+
+  return (
+    <ToolFormProvider>
+      <EsqlTool
+        mode={OnechatEsqlToolFormMode.Edit}
```

Comment:
Do we need passing mode as arg now? Depending on whether tool id is defined or not we can infer create or edit mode on a page